src/Pagination.php

```diff
@@ -0,0 +1,185 @@
+     */
+    public $forceLastKeyParam = true;
+    
+    /**
+     * @var array parameters (name => value) that should be used to obtain the current page number
+     * and to create new pagination URLs. If not set, all parameters from $_GET will be used instead.
+     *
+     * In order to add hash to all links use `array_merge($_GET, ['#' => 'my-hash'])`.
+     *
+     * The array element indexed by [[lastKeyParam]] is considered to be the current last key param (defaults to null).
+     * while the element indexed by [[pageSizeParam]] is treated as the page size (defaults to [[defaultPageSize]]).
+     */
+    public $params;
+    
+    /**
+     * This will only return 1.
+     * @return integer number of pages
+     */
+    public function getPageCount()
+    {
+        return 1;
+    }
+    
+    /**
+     * Creates the URL suitable for pagination with the specified page number.
+     * This method is mainly called by pagers when creating URLs used to perform pagination.
+     * @param string  $lastKey  The key of the last evaluated item.
+     * @param integer $pageSize The number of items on each page. If not set, the value of [[pageSize]] will be used.
+     * @param boolean $absolute Whether to create an absolute URL. Defaults to `false`.
+     * @return string the created URL
+     * @see params
+     * @see forcePageParam
+     */
+    public function createUrl($lastKey, $pageSize = null, $absolute = false)
+    {
+        $pageSize = (int) $pageSize;
+        if (($params = $this->params) === null) {
+            $request = Yii::$app->getRequest();
+            $params = $request instanceof Request ? $request->getQueryParams() : [];
+        }
+        
+        if ($lastKey !== null || $this->forcePageParam) {
+            $params[$this->lastKeyParam] = $lastKey;
+        } else {
+            unset($params[$this->lastKeyParam]);
+        }
+        if ($pageSize <= 0) {
+            $pageSize = $this->getPageSize();
+        }
+        if ($pageSize != $this->defaultPageSize) {
+            $params[$this->pageSizeParam] = $pageSize;
+        } else {
+            unset($params[$this->pageSizeParam]);
+        }
+        
+        $params[0] = $this->route === null ? Yii::$app->controller->getRoute() : $this->route;
+        $urlManager = $this->urlManager === null ? Yii::$app->getUrlManager() : $this->urlManager;
+        if ($absolute) {
+            return $urlManager->createAbsoluteUrl($params);
+        } else {
+            return $urlManager->createUrl($params);
+        }
+    }
+    
+    /**
+     * Returns just one link to the next page.
+     * @param boolean $absolute Whether the generated URLs should be absolute.
+     * @return array array containing links to the next page.
+     */
+    public function getLinks($absolute = false)
+    {
+        $pageSize = $this->getPageSize();
+        $links = [
+            Link::REL_SELF => $this->createUrl(
+                $this->getLastKey(),
```

Comment:
It will be something like this

```
            [
                ['a', 'b'],
                ['a', 'f'],
                10,
                [
                    'self' => '/index.php?r=item%2Flist&last-key%5B0%5D=a&last-key%5B1%5D=b&per-page=10',
                    'next' => '/index.php?r=item%2Flist&last-key%5B0%5D=a&last-key%5B1%5D=f&per-page=10',
                ]
            ]
```

Updated the `PaginationTest`